src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -203,53 +203,47 @@ public String getRepositoryName() {
     @Override
     public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                                    @NonNull BitbucketRepositoryProtocol protocol,
-                                   @CheckForNull Integer protocolPortOverride,
+                                   @CheckForNull String cloneLink,
                                    @NonNull String owner,
                                    @NonNull String repository) {
         switch (type) {
             case GIT:
```

Comment:
Given we are looking up and potentially passing the cloneLink from https://github.com/jenkinsci/bitbucket-branch-source-plugin/blob/3f2ca008dc5ec3b9771acd249934a8dd06a335b9/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMBuilder.java#L180 , is there any reason *not* to just use the clone link regardless of whether its HTTP or SSH as I'm guessing that is likely the be the most reliable value as it's (presumably) what is shown to BB users?

Reply:
No since it depends what have configured for regarding credentials. Whether you are using http basic auth or ssh credentials.

Reply:
See my comment https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/115#discussion_r222150736